### PR TITLE
Fixed virtual foreign key check

### PR DESCRIPTION
**Pull request by ghost**
Hi,

I am reopening PR https://github.com/phalcon/cphalcon/pull/11975 for 3.0.x branch.

### CODE REVIEW

**Comment by CrNix** [comment]
@sergeyklay I have returned the default value for _validateWithNulls_ variable, but it is set also in the for loop now.

Wouldn't it be more friendly to set it only once in the for loop? It's not used anywhere the outside of the for loop I guess, so assigning the value twice might not be that readable.

**Comment by sergeyklay** [comment]
Your PR does not make sense right now because the `validateWithNulls` has been set previously, before the loop

**Comment by CrNix** [comment]
I do not understand. That was originally the problem, when it was set before the loop.

Because when you have a model with more than one virtual foreign key, first foreign key check leaves the ```validateWithNulls``` set and might be incorrect for the next for loop iteration.

That is why I removed the default value and put the assignment inside the for loop.

**Comment by Jurigag** [comment]
He just means if relation is null then if there is enabled notNullValidations it should pass.

**Comment by sergeyklay** [comment]
Okay, could you please provide a small test which cover this case?

**Comment by CrNix** [comment]
I'll definitely give it a try.

**Comment by sergeyklay** [comment]
@CrNix I have investigated the code more details. Yes. You're right. It be more friendly to set it only once in the for loop. But in any case your PR need to be tested.

**Comment by sergeyklay** [comment]
Also could you please find the issue or open new one with script to reproduce the problem and refer this PR to the issue

**Comment by CrNix** [comment]
OK, so here is the issue with steps to reproduce this  #12071

**Comment by CrNix** [comment]
Could you explain to me please or refer me to some description what is the difference between tests and unit-tests folders?

Also, where can I define the database tables for the test? Is it OK to add them to tests/_data/schemas/mysql/mysql.dump.sql?

**Comment by sergeyklay** [comment]
> Could you explain to me please or refer me to some description what is the difference between tests and unit-tests folders?

Use always  the `tests` folder.  In the future all tests will be located in the the `tests` folder.

> Also, where can I define the database tables for the test? Is it OK to add them to tests/_data/schemas/mysql/mysql.dump.sql?

Yes. But don't forget to ammend `unit-tests/DbDescribeTest.php` ( for example https://github.com/phalcon/cphalcon/pull/12077/files). We have not had time to rewrite this test

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 - Fixed `Phalcon\Mvc\Model\Manager::getRelationRecords` to correct using multi relation column [#12035](https://github.com/phalcon/cphalcon/issues/12035)
 - Fixed `Phalcon\Acl\Resource`. Now it implements `Phalcon\Acl\ResourceInterface` [#11959](https://github.com/phalcon/cphalcon/issues/11959)
 - Fixed `save` method for all cache backends. Now it updates the `_lastKey` property correctly [#12050](https://github.com/phalcon/cphalcon/issues/12050)
+- Fixed virtual foreign key check when having multiple keys
```
Comment (sergeyklay) [inline CHANGELOG.md]: Refer to the issue please

**Comment by sergeyklay** [comment]
@CrNix Add new tables here https://github.com/phalcon/cphalcon/blob/3.0.x/tests/unit/Db/Adapter/Pdo/MysqlTest.php#L58-L81

**Comment by sergeyklay** [comment]
Thanks!

**Comment by CrNix** [comment]
You are welcome

**Pull request merged**